Update Live with current Master

rpetrusha opened the pull request:
## Update Live with current Master

BillWagner committed:
remove duplicate core version history (#9564)

* remove duplicate core version history

It's outdated, and doesn't help most customers.

* add redirect for deleted file.

* respond to feedback.

Commit:
"length" instead of "rank" (#9570)

IMHO. because "the number of elements in the initialization list" can give only length not rank.

Commit:
Revised ordinal string comparison sections (#9556)

* Initial commit

* Revised ordinal comparison sections

* Addressed feedback

* Fixed broken link

v-makoud committed:
CC86571: Adding description of 'variant' (#9543)

* CC86571: Adding description of 'variant'

Hello, @mairaw, @isaac2004, @BillWagner, @mikeblome,
This proposed file change comes from https://github.com/dotnet/docs.ja-jp/pull/200 .
Could you review this contribution and help to merge if agreed?
Many thanks in advance.

* variant interfaces --> covariant interfaces

v-makoud committed:
CC86589: Adding 'Variant' description (#9542)

* CC86589: Adding 'Variant' description

Hello, @mairaw, @BillWagner, @mikeblome,
This proposed file change comes from https://github.com/dotnet/docs.ja-jp/pull/193 .
Could you review this contribution and help to merge if agreed?
Many thanks in advance.

* variant interfaces --> contravariant interfaces

mairaw commented:
It seems this was created before the merge to live in samples was merged. Do you need to see the results of that change here?

BillWagner committed:
Fix description in inheritance article (#9574)

* update description to reflect code changes

Fixes #8567

The sample code had been updated to use auto-properties, but the descriptive text still referred to the earlier backing fields. Remove the references to the backing fields. Also, distinguish between the read-only properties, and the immutable `Author` property.

* one last edit
